modules/ensemble/lib/widget/loading_container.dart

```diff
@@ -105,6 +107,45 @@ class LoadingContainerState extends WidgetState<LoadingContainer> {
         : loadingWidget ?? const SizedBox.shrink();
   }
 
+  LinearGradient _buildGradient() {
+    switch (widget._controller.shimmerEffect) {
+      case ShimmerEffect.horizontal:
+        return LinearGradient(
+          begin: Alignment.centerLeft,
+          end: Alignment.centerRight,
+          colors: _buildGradientColors(),
+          stops: const [0.1, 0.3, 0.4],
+          tileMode: TileMode.clamp,
+        );
+      case ShimmerEffect.vertical:
+        return LinearGradient(
+          begin: Alignment.topCenter,
+          end: Alignment.bottomCenter,
+          colors: _buildGradientColors(),
+          stops: const [0.1, 0.3, 0.4],
+          tileMode: TileMode.clamp,
+        );
+      case ShimmerEffect.diagonal:
+      default:
+        return LinearGradient(
+          begin: Alignment.topLeft,
+          end: Alignment.centerRight,
+          colors: _buildGradientColors(),
+          stops: const [0.1, 0.3, 0.4],
+          tileMode: TileMode.clamp,
+        );
+    }
+  }
+
+  List<Color> _buildGradientColors() {
+    return <Color>[
+      widget._controller.baseColor ?? const Color(0xFFEBEBF4),
```

Comment:
please expose these gradient colors and stops as style properties as well so customers have full control. We will still have defaults but users should be able to overwrite.